controllers/tokens/agtokens_reconciler.go

```diff
@@ -0,0 +1,132 @@
+package tokens
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	dynatracev1alpha1 "github.com/Dynatrace/dynatrace-operator/api/v1alpha1"
+	"github.com/Dynatrace/dynatrace-operator/dtclient"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	SecretsName = "dynatrace-tokens"
```

Comment:
What happens on a token update to the ActiveGate? Does it need to restart or will it update the token internally as well?

Reply:
AG doesn't currently support tokens rotation during runtime. Restart is required.